### PR TITLE
Add Pose3-Point3 factor

### PR DESCRIPTION
**Pull request by dwisth**
This adds a new type of factor that allows for tracking 3D landmarks over time through local measurements. This is a factor that we have often looked for and wanted in the GTSAM library.

The `PoseToPointFactor` take a local measurement of a point, and computes the errors in the local coordinate system.

It passes the new unit tests for errors and Jacobians.

@mcamurri

---
This change is [Reviewable](https://reviewable.io/reviews/borglab/gtsam/359)

### CODE REVIEW

File: gtsam_unstable/slam/tests/testPoseToPointFactor.h [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  gtsam::Point3 l = gtsam::Point3(3, 0, 5);
+
+  // Factor
+  Key pose_key(1);
+  Key point_key(2);
+  SharedGaussian noise = noiseModel::Diagonal::Sigmas(Vector3(0.1, 0.1, 0.1));
+  PoseToPointFactor factor(pose_key, point_key, l_meas, noise);
+
+  // Calculate numerical derivatives
+  boost::function<Vector(const Pose3&, const Point3&)> f = boost::bind(
```
Comment (dellaert) [inline gtsam_unstable/slam/tests/testPoseToPointFactor.h:64]: auto?

Comment (dwisth) [reply gtsam_unstable/slam/tests/testPoseToPointFactor.h:64]: Done :)

File: gtsam_unstable/slam/PoseToPointFactor.h [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    return e != nullptr && Base::equals(*e, tol) && traits<Point3>::Equals(this->measured_, e->measured_, tol);
+  }
+
+  /** implement functions needed to derive from Factor */
+
+  /** vector of errors
+    * Error = pose_est.inverse()*point_est - measured_
+    * (The error is in the measurement coordinate system)
+    */
+  Vector evaluateError(const Pose3& W_T_WI,
```
[the remark this answers is not in the document]
Comment (dwisth) [reply gtsam_unstable/slam/PoseToPointFactor.h]: Done - I hope the comments are useful

**Comment by dwisth** [comment]
I think I've addressed all your comments now and ran `clang-format` to Google style. Let me know if there's anything else missing.

**Comment by dellaert** [comment]
Cool. I'll merge when CI succeeds. Thanks for your contribution!

**Comment by varunagrawal** [comment]
@dwisth can you please merge in the latest `develop` branch into this PR?
We made some updates to the CI which needs to be incorporated for the CI to pass here.

**Pull request merged**